subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/EclipseWtpPlugin.java

```diff
@@ -310,29 +352,38 @@ public void execute(EarPlugin earPlugin) {
                             }
                         });
                     }
-
                 });
             }
 
         });
-    }
 
-    private <T extends Task> void maybeAddTask(Project project, IdePlugin plugin, String taskName, Class<T> taskType, Action<T> action) {
-        if (project.getTasks().findByName(taskName) != null) {
-            return;
+        //model properties:
+        eclipseModel.getWtp().setFacet(task.map(new Transformer<EclipseWtpFacet, GenerateEclipseWtpFacet>() {
+            @Override
+            public EclipseWtpFacet transform(GenerateEclipseWtpFacet task) {
+                return task.getFacet();
+            }
+        }));
+    }
 
+    private <T extends Task> TaskProvider<T> maybeAddTask(Project project, IdePlugin plugin, String taskName, Class<T> taskType, Action<T> action) {
+        TaskContainer tasks = project.getTasks();
```

Comment:
We should remove this duplicate code, or at least simplify it in the same way as above.